src/catalog/CatalogPage.tsx

```diff
@@ -70,25 +84,35 @@ const CatalogPage = () => {
     );
   }
 
-  const totalCourses = courseData?.results?.length ?? 0;
-  const pageCount = Math.ceil((courseData?.total || totalCourses) / DEFAULT_PAGE_SIZE);
+  const totalCourses = displayData?.results?.length ?? 0;
+  const pageCount = Math.ceil((displayData?.total || totalCourses) / DEFAULT_PAGE_SIZE);
+  const hasCourses = totalCourses > 0 || (previousCourseData?.total ?? 0) > 0;
 
   return (
     <Container fluid={false} size="xl" className="pt-5.5 mb-6">
       <SubHeader
-        title={intl.formatMessage(messages.exploreCourses)}
+        title={getPageTitle({
+          intl,
+          searchString,
+          courseData,
+        })}
         className={classNames({ 'mx-2.5': isMedium })}
       />
-      {totalCourses > 0 ? (
+      {hasCourses ? (
         <>
-          <SearchField
-            key="search-field"
-            className={classNames({
-              'w-auto mx-2.5 mb-0': isMedium,
-              'mb-4 w-25': !isMedium,
-            })}
-            placeholder={intl.formatMessage(messages.searchPlaceholder)}
-          />
+          {getConfig().ENABLE_COURSE_DISCOVERY && (
+            <SearchField
+              key="search-field"
+              className={classNames({
+                'w-auto mx-2.5 mb-0': isMedium,
+                'mb-4 w-25': !isMedium,
+              })}
+              placeholder={intl.formatMessage(messages.searchPlaceholder)}
+              value={searchString}
+              onSubmit={handleSearch}
+              submitButtonLocation="external"
+            />
+          )}
           <DataTable
             isLoading={isFetching}
```

Comment:
I tried just removing this

```suggestion
```

and with the image cap loading change I like it a lot better

Using simulated slow network

https://github.com/user-attachments/assets/c84e2e7c-8213-43f3-8533-e99025a1528c

https://github.com/user-attachments/assets/ea9d439a-ad21-4a59-9a61-53a839453589

Reply:
Agree, removed